.circleci/config.yml

```diff
@@ -0,0 +1,159 @@
+version: 2.1
+
+defaults: &defaults
+  working_directory: ~/repo
+  environment:
+    LEIN_ROOT: "true"
+    JVM_OPTS: -Xmx3200m
+
+
+executors:
+  openjdk8:
+    docker:
+      - image: circleci/clojure:openjdk-8-lein-2.9.0
+    <<: *defaults
+  openjdk11:
+    docker:
+      - image: circleci/clojure:openjdk-11-lein-2.9.0
+    <<: *defaults
+
+jobs:
+  checkout_code:
+    executor: openjdk11
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v2-dependencies-{{ checksum "project.clj" }}
+            - v2-dependencies-
+      - run:
+          name: Downloading deps
+          command: |
+            lein deps
+      - save_cache:
+          paths:
+            - ~/.m2
+          key: v2-dependencies-{{ checksum "project.clj" }}
+      - persist_to_workspace:
+          root: /home/circleci
+          paths:
+            - .m2
+            - repo
+
+  lint_code:
+    executor: openjdk11
+    steps:
+      - attach_workspace:
+          at: /home/circleci
+      - run:
+          name: Install make
+          command: |
+            sudo apt-get install make
+      - run:
+          name: Running Eastwood
+          command: |
+            make eastwood
+      - run:
+          name: Running cljfmt
+          command: |
+            make cljfmt
+
+  test_code:
+    description: Run tests
+    parameters:
+      jdk_version:
+        description: Version of JDK to test against
+        type: string
+      clojure_version:
+        description: Version of Clojure to test against
+        type: string
+    executor: << parameters.jdk_version >>
+    environment:
+      VERSION: << parameters.clojure_version >>
+    steps:
+      - attach_workspace:
+          at: /home/circleci
+      - run:
+          name: Install make
+          command: |
+            sudo apt-get install make
+      - run:
+          name: Running tests
+          command: make test
+
+  codecov:
+    executor: openjdk11
+    environment:
+      VERSION: "1.9"
+    steps:
+      - attach_workspace:
+          at: /home/circleci
+      - run:
+          name: Install make
+          command: |
+            sudo apt-get install make
+      - run:
+          name: Running cloverage
+          command: |
+            make cloverage
+
+workflows:
+  version: 2.1
+  build_and_test:
```

Comment:
Not sure what a `workflow` is exactly in CircleCI, but as we don't really do any explicit "build", I think the name is a bit confusing. I guess the real stages of the build process are "Linting" and "Running tests".

Reply:
The `build_and_test` refers to the overall process/workflow (i.e. linting + testing). Happy to take suggestions on the right name.